src/__integrations__/pmClient.addWallet.integration.test.ts

```diff
@@ -0,0 +1,190 @@
+    expect(
+      (
+        await pmClient.addWalletUsingPOST({
+          Bearer: 'Bearer ' + sessionToken,
+          walletRequest: {
+            data: {},
+          },
+          language: 'it',
+        })
+      ).fold(
+        err => (err.pop()?.value as Response).status,
+        myRes => myRes.status,
+      ),
+    ).toEqual(422);
+  });
+
+  it('should return 200 when the walletRequest reflects the inputs collected during the payment flow', async () => {
+    expect(
+      (
+        await pmClient.addWalletUsingPOST({
+          Bearer: 'Bearer ' + sessionToken,
+          walletRequest: {
+            data: {
+              type: TypeEnum.CREDIT_CARD,
+              creditCard: {
+                brand: 'VISA',
+                expireMonth: '03',
+                expireYear: '25',
+                holder: 'Ciccio Mio',
+                pan: '4024007182788397',
+                securityCode: '666',
+              },
+              favourite: true,
+              idPagamentoFromEC: '2a4afce9-40e5-4be7-885b-a5223ff73e9a', // needs to exist
+            },
+          },
+          language: 'it',
+        })
+      ).fold(
+        err => (err.pop()?.value as Response).status,
+        myRes => myRes.status,
+      ),
+    ).toEqual(200);
+  });
+
+  it('should return 200 when the walletRequest contains the bare minimum fields', async () => {
+    expect(
+      (
+        await pmClient.addWalletUsingPOST({
+          Bearer: 'Bearer ' + sessionToken,
+          walletRequest: {
+            data: {
+              type: TypeEnum.CREDIT_CARD,
+              creditCard: {
+                expireMonth: '03',
+                expireYear: '25', // year must be encoded with two digits
+                holder: 'Ciccio Mio',
+                pan: '4024007182788397',
+              },
+              idPagamentoFromEC: '', // needs to exist
+            },
+          },
+          language: 'it',
+        })
+      ).fold(
+        err => (err.pop()?.value as Response).status,
+        myRes => myRes.status,
+      ),
+    ).toEqual(200);
+  });
+
+  it('should return 422 when the credit card expiring year is encoded with 4 digits', async () => {
+    expect(
+      (
+        await pmClient.addWalletUsingPOST({
+          Bearer: 'Bearer ' + sessionToken,
+          walletRequest: {
+            data: {
+              type: TypeEnum.CREDIT_CARD,
+              creditCard: {
+                expireMonth: '03',
+                expireYear: '2025', // year must be encoded with two digits
+                holder: 'Ciccio Mio',
+                pan: '4024007182788397',
+              },
+              idPagamentoFromEC: '2a4afce9-40e5-4be7-885b-a5223ff73e9a', // needs to exist
+            },
+          },
+          language: 'it',
+        })
+      ).fold(
+        err => (err.pop()?.value as Response).status,
+        () => undefined,
+      ),
+    ).toEqual(422);
+  });
+});
```

Comment:
Does't miss a test on answer response fields ?

Reply:
@pasqualespica I agree. I'll add it